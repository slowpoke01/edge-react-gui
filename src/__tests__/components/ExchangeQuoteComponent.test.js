/* globals describe it expect */
/* eslint-disable flowtype/require-valid-file-annotation */

import * as React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import { getTheme } from '../../components/services/ThemeContext.js'
import { ExchangeQuoteComponent } from '../../components/themed/ExchangeQuoteComponent.js'

describe('ExchangeQuote', () => {
  it('should render with loading props', () => {
    const renderer = new ShallowRenderer()

    const props = {
      isTop: true | undefined,
      walletIcon: 'string',
      cryptoAmount: 'string',
      currency: 'string',
      currencyCode: 'string',
      fiatCurrencyCode: 'string',
      fiatCurrencyAmount: 'string',
      walletName: 'string',
      total: 'string',
      miningFee: 'string' | undefined,
      theme: getTheme()
    }
    const actual = renderer.render(<ExchangeQuoteComponent {...props} />)

    expect(actual).toMatchSnapshot()
  })
})
