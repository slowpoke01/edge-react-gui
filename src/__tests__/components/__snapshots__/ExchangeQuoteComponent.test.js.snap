// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeQuote should render with loading props 1`] = `
<WithTheme(CardComponent)>
  <WithTheme(CardContentComponent)
    image={
      <FastImage
        source={
          Object {
            "uri": "string",
          }
        }
        style={
          Object {
            "height": 34,
            "resizeMode": "contain",
            "width": 34,
          }
        }
      />
    }
    subTitle="string"
    subValue="string string"
    title="string"
    value="string string"
  />
  <React.Fragment>
    <WithTheme(DataRowComponent)
      label={
        <WithTheme(EdgeTextComponent)
          style={
            Object {
              "fontSize": 17,
              "fontWeight": "600",
            }
          }
        >
          Max network fee
        </WithTheme(EdgeTextComponent)>
      }
      marginRem={
        Array [
          1,
          0,
          0,
        ]
      }
      value={
        <WithTheme(EdgeTextComponent)
          style={
            Object {
              "fontSize": 17,
              "fontWeight": "600",
            }
          }
        >
          0
        </WithTheme(EdgeTextComponent)>
      }
    />
    <WithTheme(DataRowComponent)
      label={
        <WithTheme(EdgeTextComponent)
          style={
            Object {
              "fontSize": 17,
              "fontWeight": "600",
            }
          }
        >
          Total Amount
        </WithTheme(EdgeTextComponent)>
      }
      value={
        <WithTheme(EdgeTextComponent)
          style={
            Object {
              "fontSize": 17,
              "fontWeight": "600",
            }
          }
        >
          string string
        </WithTheme(EdgeTextComponent)>
      }
    />
  </React.Fragment>
</WithTheme(CardComponent)>
`;
